src/display-buffer.coffee

```diff
@@ -841,9 +841,15 @@ class DisplayBuffer extends Model
     if screenLine.isSoftWrapped() and column >= maxScreenColumn
       if wrapAtSoftNewlines
         row++
-        column = 0
+        column = @screenLines[row].clipScreenColumn(0)
       else
         column = screenLine.clipScreenColumn(maxScreenColumn - 1)
+    else if screenLine.isColumnInsidePhantomToken(column)
+      if wrapAtSoftNewlines
```

Comment:
> However, it breaks some existing specs that I think might actually be wrong. The are all specs for clipping the position at the beginning of a continuation of a soft-wrapped line...

Yes, as mentioned in [another comment](https://github.com/atom/atom/pull/5567#discussion_r25004660) the reason why I came up with such behavior (and its related specs) was that I wanted the cursor to bump to the first *real* character of the line whenever a user clicked on the empty space (aka phantom token) of an indented soft wrapped line, or when moving down (see previous comment for a better explanation of this issue). Assuming this behavior is correct, it seemed reasonable enough to change `moveLeft` to pass a parameter when clipping screenPosition, rather than changing both `moveDown` and the click handler.